src/unity/toolkits/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -0,0 +1,171 @@
+#include <unity/toolkits/one_shot_object_detection/one_shot_object_detector.hpp>
+
+#include <unity/toolkits/object_detection/object_detector.hpp>
+
+// TODO: Clean up imports.
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <limits>
+#include <numeric>
+#include <queue>
+#include <utility>
+#include <vector>
+#include <random>
+
+#include <logger/assertions.hpp>
+#include <logger/logger.hpp>
+#include <random/random.hpp>
+
+#include <boost/gil/gil_all.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/utilities.hpp>
+
+#include <image/numeric_extension/perspective_projection.hpp>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+using turi::coreml::MLModelWrapper;
+using namespace Eigen;
+using namespace boost::gil;
+
+namespace turi {
+namespace one_shot_object_detection {
+
+/* A ParameterSweep class to randomly generate different samples of parameters
+ * that can later be used to compute the transformation matrix necessary to 
+ * create image projections.
+ */
+class ParameterSweep {
+public:
+  ParameterSweep(int width, int height) {
+    width_ = width;
+    height_ = height;
+  }
+
+  double deg_to_rad(double angle) {
+    return angle * M_PI / 180.0;
+  }
+
+  double get_theta() {
+    return deg_to_rad(theta_);
+  }
+
+  double get_phi() {
+    return deg_to_rad(phi_);
+  }
+
+  double get_gamma() {
+    return deg_to_rad(gamma_);
+  }
+
+  int get_dz() {
+    return dz_;
+  }
+
+  double get_focal() {
+    return focal_;
+  }
+
+  void sample(long seed) {
+    /* Barebones */
+    theta_ = 0.0;
+    phi_ = 0.0;
+    gamma_ = 0.0;
+    std::normal_distribution<double> focal_distribution((double)width_, focal_stdev_);
+    focal_generator_.seed(seed+6);
+    focal_ = focal_distribution(focal_generator_);
+    dz_ = focal_;
+  }
+
+private:
+  int width_;
+  int height_;
+  double focal_stdev_ = 40.0;
+  std::default_random_engine focal_generator_;
+  double theta_;
+  double phi_;
+  double gamma_;
+  int dz_;
+  double focal_;
+
+};
+
+gl_sframe _augment_data(gl_sframe data, gl_sframe backgrounds, long seed) {
```

Comment:
There is no need to prefix functions with `_`. Prefer to use namespaces in C++. For file-internal helper functions, use anonymous namespaces (or the (older) `static` keyword)

Reply:
Done.